common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -44,6 +44,29 @@ class VerticalBlock(SequenceFields, XModuleFields, StudioEditableBlock, XmlParse
 
     show_in_read_only_mode = True
 
+    def student_view_data(self, context):
+        """
+        Returns JSON data for the student view.
+        """
+        children_data = []
+        for child_block in self.get_display_items():
+            child_fragment = child_block.render(STUDENT_VIEW, context)
+            children_data.append({
+                'usage_id': six.text_type(child_block.location),
+                'block_type': child_block.location.block_type,
+                'display_name': child_block.display_name,
+                'fragment': child_fragment.to_dict(),
+                'completion': '<completion_value>'
+            })
+
+        return {
+            'usage_id': six.text_type(self.location),
+            'block_type': self.location.block_type,
+            'display_name': self.display_name_with_default,
```

Comment:
I believe that `student_view_data` doesn't usually include `usage_id`, `block_type`, or `display_name` since they are [available in the course blocks API as fields that are peers of the `student_view_data` field](https://openedx.atlassian.net/wiki/spaces/EDUCATOR/pages/29688043/Course+Blocks+API). So including them in `student_view_data` is redundant for the course blocks API, and also doesn't feel right to me here.